jbpm-task-assigning/jbpm-task-assigning-core/src/test/java/org/jbpm/task/assigning/model/solver/TestSolver.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.task.assigning.model.solver;
+
+import java.util.List;
+
+import org.jbpm.task.assigning.AbstractTaskAssigningCoreTest;
+import org.jbpm.task.assigning.model.Task;
+import org.jbpm.task.assigning.model.TaskAssigningSolution;
+import org.jbpm.task.assigning.model.User;
+import org.junit.Test;
+import org.optaplanner.core.api.solver.Solver;
+
+import static org.jbpm.task.assigning.TestDataSet.SET_OF_24TASKS_8USERS_SOLUTION;
+import static org.jbpm.task.assigning.model.solver.TaskHelper.extractTaskList;
+import static org.jbpm.task.assigning.model.solver.TaskHelper.isPotentialOwner;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class TestSolver extends AbstractTaskAssigningCoreTest {
+
+    private static final long MILLISECONDS_TIME_SPENT_LIMIT = 20000;
+    private static final long TEST_TIMEOUT = MILLISECONDS_TIME_SPENT_LIMIT + 3000;
+
+    @Test(timeout = TEST_TIMEOUT)
+    public void startSolverAndSolution24Tasks8Users() throws Exception {
+        testSolverStartAndSolution(MILLISECONDS_TIME_SPENT_LIMIT, SET_OF_24TASKS_8USERS_SOLUTION.resource(), 20);
+    }
+
+    /**
+     * Tests that solver for the tasks assigning problem definition can be properly started, a solution can be produced,
+     * and that some minimal constrains are met by de solution.
+     */
+    private void testSolverStartAndSolution(long millisecondsSpentLimit, String solutionResource, int expectedSteps) throws Exception {
+        Solver<TaskAssigningSolution> solver = createNonDaemonSolver(millisecondsSpentLimit);
```

Comment:
Not sure if we understood each other correctly. I meant using step count termination [1] instead of `millisecondsSpentLimit` and adding a timeout just to be sure the test never hangs.

The timeout is there, but the solver is still configured to terminate after a certain period of time.

[1] https://docs.optaplanner.org/latest/optaplanner-docs/html_single/#stepCountTermination